state/application.go

```diff
@@ -272,10 +272,6 @@ func (a *Application) removeOps(asserts bson.D) ([]txn.Op, error) {
 			Id:     a.doc.DocID,
 			Assert: asserts,
 			Remove: true,
-		}, {
-			C:      settingsC,
-			Id:     a.settingsKey(),
-			Remove: true,
```

Comment:
This operation gets added, tracing through to appCharmDecRefOps(...), at https://github.com/juju/juju/blob/develop/state/charmref.go#L80.

If we are to keep the remove settings operation I suggest we get rid of, we end up wtih 2 remove settings operations  that are exactly the same. I have added my observations into the pastebin :)